doc/builders/special/incremental-build.section.md

```diff
@@ -0,0 +1,20 @@
+# pkgs.buildIncremental.*  {#sec-incremental-build}
+
+`pkgs.buildIncremental` provides a way to build derivations incrementally. It consists of two functions to make incremental builds using nix possible.
+**Attention** Incrementality can't be added in a transparent manner, so incrementality is added within the derivation paradigm. Nix is not able to manage the state, but the underlying build tool can delegate the state management to recursive nix.
+
+## Building a project incrementally 
```

Comment:
This documentation needs to be clearer about how it can be used. It should come with an example showing users how it can be set up for a specific nixpkgs package, with code and commands that can be run and showing how much faster the build becomes.

Reply:
i've added the example from the code.